ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -718,4 +719,18 @@ default ColumnInfo getColumnInfo(org.apache.hadoop.hive.ql.metadata.Table hmsTab
             "for a specific column.");
   }
 
+  default boolean supportsMetadataDelete() {
+    return false;
+  }
+
+  default boolean canPerformMetadataDelete(org.apache.hadoop.hive.ql.metadata.Table hmsTable, SearchArgument searchArgument) {
+    throw new UnsupportedOperationException("Storage handler does not support validation of metadata " +
+            " delete operation using search argument.");
+  }
+
+  default void performMetadataDelete(org.apache.hadoop.hive.ql.metadata.Table hmsTable, SearchArgument searchArgument) {
```

Comment:
should if be simply `delete ` and each StorageHandler should apply the proper strategy?

Reply:
This function is not required now. Completely removed.